src_c/draw.c

```diff
@@ -1079,48 +1101,37 @@ draw_aaline(SDL_Surface *surf, Uint32 color, float from_x, float from_y, float t
         return;
     }
 
-    if (fabs(dx) > fabs(dy)) {
+    if (fabs(dx) >= fabs(dy)) {
         if (from_x > to_x) {
             SWAP(from_x, to_x, swaptmp)
             SWAP(from_y, to_y, swaptmp)
             dx = -dx;
             dy = -dy;
         }
         slope = dy / dx;
-        // 1. Draw start of the segment
-        pt_x = trunc(from_x) + 0.5; /* This makes more sense than trunc(from_x+0.5) */
-        pt_y = from_y + slope * (pt_x - from_x);
-        xgap = INVFRAC(from_x);
-        ifrom_x = (int)pt_x;
-        ifrom_y = (int)pt_y;
-        yf = pt_y + slope;
-        brightness1 = INVFRAC(pt_y) * xgap;
-        brightness2 = FRAC(pt_y) * xgap;
-        pixel = surf_pmap + pixx * ifrom_x + pixy * ifrom_y;
-        DRAWPIX32(pixel, colorptr, brightness1, blend)
-        pixel += pixy;
-        DRAWPIX32(pixel, colorptr, brightness2, blend)
-        // 2. Draw end of the segment
-        pt_x = trunc(to_x) + 0.5;
-        pt_y = to_y + slope * (pt_x - to_x);
-        xgap = FRAC(to_x); /* this also differs from Hugo's description. */
-        ito_x = (int)pt_x;
-        ito_y = (int)pt_y;
-        brightness1 = INVFRAC(pt_y) * xgap;
-        brightness2 = FRAC(pt_y) * xgap;
-        pixel = surf_pmap + pixx * ito_x + pixy * ito_y;
-        DRAWPIX32(pixel, colorptr, brightness1, blend)
-        pixel += pixy;
-        DRAWPIX32(pixel, colorptr, brightness2, blend)
+        G_x = ceil(from_x);
```

Comment:
Nice. Still not sure about this: Why use ceil rather than round (or trunc(x+0.5))? Won't that always round to the right or bottom pixel instead of the nearest pixel?